tests/components/nextbus/test_sensor.py

```diff
@@ -0,0 +1,282 @@
+"""The tests for the nexbus sensor component."""
+from unittest.mock import patch
+from copy import deepcopy
+
+import pytest
+
+import homeassistant.components.sensor as sensor
+import homeassistant.components.nextbus.sensor as nextbus
+
+from tests.common import (assert_setup_component,
+                          async_setup_component,
+                          MockDependency)
+
+
+VALID_AGENCY = 'sf-muni'
+VALID_ROUTE = 'F'
+VALID_STOP = '5650'
+VALID_AGENCY_TITLE = 'San Francisco Muni'
+VALID_ROUTE_TITLE = 'F-Market & Wharves'
+VALID_STOP_TITLE = 'Market St & 7th St'
+SENSOR_ID_SHORT = 'sensor.sf_muni_f'
+
+CONFIG_BASIC = {
+    'sensor': {
+        'platform': 'nextbus',
+        'agency': VALID_AGENCY,
+        'route': VALID_ROUTE,
+        'stop': VALID_STOP,
+    }
+}
+
+CONFIG_INVALID_MISSING = {
+    'sensor': {
+        'platform': 'nextbus',
+    }
+}
+
+
+async def assert_setup_sensor(hass, config, count=1):
+    """Set up the sensor and assert it's been created."""
+    with assert_setup_component(count):
+        assert await async_setup_component(hass, sensor.DOMAIN, config)
+
+
+@pytest.yield_fixture
+def mock_validate_tags():
+    """Create a mock validate_tags function."""
+    with patch('homeassistant.components.nextbus.sensor.validate_tags') \
+            as validate_tags:
+
+        yield validate_tags
+
+
+@pytest.yield_fixture
+def mock_predictions():
+    """Create a mock get_predictions function."""
+    with patch('homeassistant.components.nextbus.sensor.get_predictions') \
```

Comment:
Don't patch the platform code. Patch the client method that returns the predictions.